infra/base-images/base-builder/jcc/jcc.go

```diff
@@ -165,15 +166,72 @@ func CorrectMissingHeaders(bin string, cmd []string) (bool, error) {
 	return false, nil
 }
 
-func compile(bin string, args []string) (int, string, string) {
-	cmd := exec.Command(bin, args...)
+func RunCommand(cmd *exec.Cmd) (int, string, string) {
+	// Executes a command and returns the output.
 	var outb, errb bytes.Buffer
 	cmd.Stdout = &outb
 	cmd.Stderr = &errb
 	cmd.Run()
 	return cmd.ProcessState.ExitCode(), outb.String(), errb.String()
 }
 
+func ExecOriginalCommand(bin string, args []string) (int, string, string) {
+	// Executes the original command.
+	cmd := exec.Command(bin, args...)
+	return RunCommand(cmd)
+}
+
+func Contains(slice []string, item string) bool {
+	// Checks if the slice contains item.
+	for _, s := range slice {
+		if strings.EqualFold(s, item) {
+			return true
+		}
+	}
+	return false
+}
+
+func FindTargetFile(args []string) string {
```

Comment:
What's the behaviour when there's multiple .c/.cxx files passed to one command? 

We may want to try to get separate files for these right?

Reply:
> What's the behaviour when there's multiple .c/.cxx files passed to one command?

I tested this on a manual example with 2 files:
```bash
root@cdf5003b0ce7:/src# cat file1.c
int main1(){
        int a = 1;
        return a;
}
root@cdf5003b0ce7:/src# cat file2.c
int main(){
        int b = 2;
        return b;
}
root@cdf5003b0ce7:/src# $CXX file1.c file2.c
```
This will generate one file (`file1.c.ast`) that concatenates the ASTs of both `file1.c` and `file2.c`.

> We may want to try to get separate files for these right?

Yes, we can separate them via the following script (also assuming 2 files, but easily expandable to multiple):
```python
with open('both.ast', 'r') as f:
    data = f.read()

# Find the position where the first file ends and the second file begins
split_pos = data.find('}{') + 1

# Write the data back into two separate files
with open('file1.ast', 'w') as f:
    f.write(data[:split_pos])

with open('file2.ast', 'w') as f:
    f.write(data[split_pos:])
```

For each function, I should be able to identify their belonging fIlename from each AST's `'loc'` filed:
```python
>>> [datum['loc'] for datum in data['inner'] if datum['id']=='0x6a3e030']
[{'offset': 4, 'file': 'file1.c', 'line': 1, 'col': 5, 'tokLen': 5}]
```

Would this solution be too hacky?

Reply:
This does seem a bit too hacky. We may want to instead invoke clang multiple times per matching source file we find in the commandline.

Reply:
OK that sounds doable. I will re-execute the same command multiple times and only include one of the C/CPP files at a time.

Would it be possible that one C/CPP file cannot build without others?
I am not sure if we can assume the first AST is always the first C/CPP file in the command line, if we include multiple files and reorder them at each time.

Reply:
Not sure if it's possible that a C/CPP file cannot build without others being available in the cmdline. Also these invocation is not actually building, just simply parsing right? 

Can you explain a bit more what you mean by "I am not sure if we can assume the first AST is always the first C/CPP file in the command line, if we include multiple files and reorder them at each time." ? 

Can we not generate filename.cpp.AST filename2.cpp.AST etc ?

Reply:
> Not sure if it's possible that a C/CPP file cannot build without others being available in the cmdline.

I am probably overthinking this, but just in case: Say the original command is `clang-jcc file1.c file2.c`, and we want to split it into `clang-jcc file1.c > file1.c.ast` and `clang-jcc file2.c > file2.c.ast` so that we can obtain two ASTs separately. Would it be possible that some functions/structures referenced in `file1.c` were defined in `file2.c`, hence `clang-jcc file1.c > file1.c.ast` fails to generate the AST of those functions/structures?

> Can you explain a bit more what you mean

Sorry, please don't worry about it.
I was thinking about an alternative solution, but that does not make sense.

Reply:
OK, I will implement the multi-file solution for now.
If it does not work for some rare cases, we can still revert to my hacky solution.

Reply:
Done :)